pallets/allocations/src/lib.rs

```diff
@@ -185,13 +217,25 @@ pub mod pallet {
 	}
 
 	#[pallet::storage]
-	#[pallet::getter(fn oracles)]
-	pub type Oracles<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+	pub(crate) type StorageVersion<T: Config> = StorageValue<_, Releases, ValueQuery>;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	#[pallet::storage]
+	#[pallet::getter(fn validator_set)]
+	pub type ValidatorSet<T: Config> = StorageValue<_, BoundedVec<T::AccountId, benchmarking::MaxMembers>, ValueQuery>;
 }
 
 impl<T: Config> Pallet<T> {
 	pub fn is_oracle(who: T::AccountId) -> bool {
-		Self::oracles().contains(&who)
+		#[cfg(feature = "runtime-benchmarks")]
+		if <ValidatorSet<T>>::get().is_empty() {
+			T::OracleMembers::contains(&who)
+		} else {
+			Self::contains(&who)
+		}
+
+		#[cfg(not(feature = "runtime-benchmarks"))]
+		T::OracleMembers::contains(&who)
```

Comment:
Suggest a simplification like:
```suggestion
		#[cfg(feature = "runtime-benchmarks")]
		return T::OracleMembers::contains(&who) || Self::benchmark_oracles().contains(&who);
		#[cfg(not(feature = "runtime-benchmarks"))]
		return T::OracleMembers::contains(&who);
```